docs/commerce/checkout/customize_checkout.md

```diff
@@ -0,0 +1,202 @@
+
+The controller contains a Symfony form that collects user selections. 
+It can reuse fields and functions that come from the checkout component, for example, 
+after you check whether the form is valid, use the `advance` method to go to the next stage 
+of the process.
+
+``` php hl_lines="23 24"
+[[= include_file('code_samples/front/shop/checkout/src/Controller/SelectSeatStepController.php') =]]
+```
+
+### Create Twig template
+
+You also need a Twig template to render the Symfony form.
+In `templates/themes/custom/storefront/checkout/step`, create a layout that uses JavaScript to translate clicking into a grid to a change in value:
+
+```html+twig
+[[= include_file('code_samples/front/shop/checkout/templates/themes/storefront/checkout/step/select_seat.html.twig') =]]
+```
+
+### Configure repository
+
+At this point, you must inform the application about a repository that the workflow supports.
+
+You do it repository configuration, by replacing the `ibexa_checkout` configuration with one for `seat_selection_checkout`:
+
+``` yaml
+ibexa:
+    repositories:
+        default: 
+            checkout:
+                workflow: seat_selection_checkout
+```
+
+### Restart application
+
+You are now ready to see the results of your work.
+Shut down the application, clear browser cache, and restart the application.
+You should be able to see a different checkout applied after you have added products to a cart.
+
+## Create a single-form checkout
+
+Another way of customizing the process would be to implement a single-form checkout.
+Such solution could work for certain industries, where simplicity is key.
+The single form's basic advantage is simplified navigation with less clicks to complete the transaction.
+
+### Define workflow
+
+To create a single-form checkout, define a workflow that has two stages, `initialized` and `completed`, and one transition, from `initialized` or `completed` to `completed`.
+
+``` yaml hl_lines="3 18 19"
+[[= include_file('code_samples/front/shop/checkout/config/packages/ibexa.yaml', 25, 27) =]] [[= include_file('code_samples/front/shop/checkout/config/packages/ibexa.yaml', 84, 103) =]] [[= include_file('code_samples/front/shop/checkout/config/packages/ibexa.yaml', 23, 24) =]]
+```
+
+### Create controller
+
+Add a controller in project code that is a regular Symfony controller, which reuses classes provided by the application.
+Within the controller, create a form that contains all the necessary fields, such as the shipping and billing addresses, as well as shipping and billing methods.
+
+In the `src/Controller/Checkout` folder, create a file that resembles the following example:
+
+``` php
+[[= include_file('code_samples/front/shop/checkout/src/Controller/SinglePageCheckout.php') =]]
+```
+
+Again, after you handle the form and validate it, you advance the workflow by using the `advance` method provided in the base component.
```

Comment:
I would not use "again" here, but repeat the explanation from above. This is a separate procedure than the one above and we can't assume readers will read the first one before starting with this one.